pkg/driver/deployment/knative.go

```diff
@@ -83,11 +87,63 @@ func knativeDeploySingleFunction(function *common.Function, yamlPath string, isP
 		// second, then round to an integer as that is what the knative config expects
 	}
 
+	// Default is the function name, but if the function is a vSwarm function, the proxy name is used
+	proxyName := function.Name
+
+	if vSwarm {
+		// Read and unmarshal the mapper output file into a map
+
+		mapperFile, err := os.ReadFile("tools/mapper/output.json")
+		if err != nil {
+			log.Warn("No mapper output file")
+		}
+		var mapperOutput map[string]map[string]string // HashFunction mapped to vSwarm function yaml.
+		err = json.Unmarshal(mapperFile, &mapperOutput)
+		if err != nil {
+			log.Warn("Failed to unmarshal mapper output file")
+		}
+
+		// Find the proxy name for the function
+		proxyName = mapperOutput[function.HashFunction]["proxy-function"]
+		log.Info("Proxy name: ", proxyName)
+		// Read the deployment info file for yaml locations and predeployment commands if any
+
+		deploymentInfoFile, err := os.ReadFile(yamlPath + "deploy_info.json")
```

Comment:
This file seems important for understanding the code. Can you pull it out of the tar.gz?